src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -60,7 +61,9 @@ public Object create(X x) {
 
   protected static Map<String,Boolean> getPermissionMap(final X x) {
     Session             session = x.get(Session.class);
-    Map<String,Boolean> map     = (Map) session.getContext().get(CACHE_KEY);
+    Map<String,Boolean> map = session.getUserId() == ((User) x.get("user")).getId() ? 
```

Comment:
Document why you're doing this.
But if you don't use the cache map this will create a new one and override the one in the cache. Is that what you want?